src/transport/GroupSession.h

```diff
@@ -57,7 +57,8 @@ class IncomingGroupSession : public Session, public ReferenceCounted<IncomingGro
         return subjectDescriptor;
     }
 
-    bool RequireMRP() const override { return false; }
+    bool AllowsMRP() const override { return false; }
+    bool IsTransportTCP() const override { return false; }
```

Comment:
IsTransportTCP does not seem used but is defined. Could we drop it from this PR? We seem to only need `AllowsMRP`

Reply:
The PeerAddress being of type TCP is one of the key flags that the underlying session establishment code would use to Connect over TCP, or use MRP. This function essentially returns that,(similar to what AllowsMRP does). AllowsMRP() is checking for UDP transport. So, essentially, it is the same as a IsTransportUDP(), isn't it? We need to associate the TCP connection with the session as we are not sharing sessions between MRP and TCP. Although this API is not being used in this PR, it would be used in subsequent changes.

Reply:
In particular this PR does not make use of IsTransportTCP so I have a hard time figuring out naming or usage suggestions.

I would be leaning towards a `SupportsLargePayloads` or `GetMaxPayloadSize` and `CloseSessionOnDisconnect` or similar items instead of a `if tcp` however I cannot check if my suggestions are sane without seeing usages of this IsTransportTCP. So I am requesting to remove this method and add it if really needed and used later on (and in those case to strongly consider similar naming of `Allow*` without passing in transport types up the chain outside transport.

Reply:
@andy31415, renamed the IsTransportTCP() to AllowsLargePayload().